tensorflow/core/kernels/mkl_matmul_op.cc

```diff
@@ -161,8 +161,19 @@ class MklMatMulOp : public OpKernel {
     VLOG(2) << "MKL DNN SGEMM called";
 #ifdef ENABLE_MKLDNN_THREADPOOL
     MklDnnThreadPool eigen_tp(ctx);
-    sgemm(char_transa, char_transb, m, n, k, alpha, a, lda, b, ldb, beta, c,
-          ldc, &eigen_tp);
+    // With threadpool , the runtime overhead is comparable to the kernel
+    // execution for small kernel sizes. For such sizes, it may be better to run
+    // the kernel single threaded. Here we are coming up with a cost model based
+    // on L1 sizes. If we find that matrices are small enough, we will execute
+    // single threaded. This may need tuning.
+    if (ExecuteSingleThreadedGemm(m, n, k)) {
```

Comment:
if (ExecuteSingleThreadedGemm(m, n, k)) {
    MklDnnThreadPool eigen_tp(ctx, 1);
    sgemm(char_transa, char_transb, m, n, k, alpha, a, lda, b, ldb,beta, c, ldc, &eigen_tp);
} else {
    MklDnnThreadPool eigen_tp(ctx);
    sgemm(char_transa, char_transb, m, n, k, alpha, a, lda, b, ldb,beta, c, ldc, &eigen_tp);
}

Would use this ?

Reply:
dnnl_sgemm(char_transa, char_transb, m, n, k, alpha, a, lda, b, ldb, beta, c, ldc); will use all cores to compute sgemm, like the following #else part.